assets/src/web-stories-block/block/block-types/selected-stories/fetchSelectedStories.js

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Notice, Placeholder } from '@wordpress/components';
+import { BlockIcon } from '@wordpress/block-editor';
+import { useEffect, useState } from '@wordpress/element';
+import apiFetch from '@wordpress/api-fetch';
+import { addQueryArgs } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import { useConfig } from '../../config';
+import LoaderContainer from './components/loaderContainer';
+
+const LoadingPlaceholder = styled(Placeholder)`
+  &.is-appender {
+    min-height: 0;
+    margin-top: 20px;
+  }
+`;
+
+const ErrorNotice = styled(Notice)`
+  width: 100%;
+`;
+
+function FetchSelectedStories({
+  icon,
+  label,
+  selectedStories,
+  setSelectedStoriesObject,
+  setIsFetchingSelectedStories,
+}) {
+  const { api } = useConfig();
+
+  const [errorMsg, setErrorMsg] = useState('');
+  const placeholderIcon = <BlockIcon icon={icon} showColors />;
+
+  const fetchStories = async () => {
+    try {
+      const response = await apiFetch({
+        path: addQueryArgs(api.stories, {
+          _embed: 'author',
+          _web_stories_envelope: true,
```

Comment:
Why is `_web_stories_envelope` used here?

Reply:
I believe this was done for consistency, as other data providers also envelop the response data, this gives consistent use of `response.body` across. However not enveloping this would also have worked by directly accessing the response.

Reply:
The reason I implemented `_web_stories_envelope` in the first place was we could easily access headers, but if you are not using the headers for anything, then i would remove it.

Reply:
removed.